src/pages/[id].page.jsx

```diff
@@ -24,24 +24,40 @@ export default function Details({ details }) {
 
   return (
     <Layout pageTitle={componentName}>
-      <div className="details-page">
-        <DetailsBanner name={componentName} image={image} />
+      <div className="details-page__header">
+        <div className="details-page__primary">
+          <DetailsBanner name={componentName} image={image} />
+        </div>
+      </div>
+      <div className="details-page__primary">
         <Definition>
           <p>{definition}</p>
         </Definition>
-        <UsageGuidelines>
-          <RichText markdown={usageGuidelines} />
-        </UsageGuidelines>
+      </div>
+      <div className="details-page__banner">
+        <div className="details-page__secondary">
+          <UsageGuidelines>
+            <RichText markdown={usageGuidelines} />
+          </UsageGuidelines>
+        </div>
+      </div>
+      <div className="details-page__primary">
         <div className="cmp-specifications-block">
           {specificationBlocks.map((block) => (
             <SpecificationBlock key={block.id} heading={block.heading}>
               <RichText markdown={block.content} />
             </SpecificationBlock>
           ))}
         </div>
-        {/* List of links to other components */}
-        <RelatedComponents components={relatedComponents} />
       </div>
+      {/* List of links to other components. Only renders when it contains content. */}
+      {!relatedComponents && (
```

Comment:
Does this logic need to be here since it's being handled in the RelatedComponents component?

Reply:
Right, this was the weird problem I was having. I wanted to either render <RelatedComponents> with the margin/padding divs around it, or render neither the divs nor the component. When the logic was handled in RelatedComponents, it would be able to remove itself from the DOM, but not the wrappers, so the margins & background color were still there.

Reply:
I ended up removing the logic from the component and adding it here instead to get around that. Unfortunately, that broke a RelatedComponents test, so now I have to refactor it for the [id] page 😓